### PR TITLE
Adding APIRule create troubleshooting guide

### PR DESCRIPTION
**Pull request by videlov**
**Description**

Changes proposed in this pull request:

- Adding APIRule create troubleshooting guide

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/15115

### CODE REVIEW

File: docs/04-operation-guides/troubleshooting/api-exposure/apix-10-api-rule-troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+```
+{"code":"ERROR","desc":"Validation error: Attribute \".spec.rules[0].accessStrategies.accessStrategies[0].handler\": allow access strategy is not allowed in combination with other access strategies"}
+```
+
+## Remedy
+
+You should decide to `allow` access to the specific path or restrict it via `jwt` security token. Both at the same time is not allowed.
+
+---
+## Issue - Using Istio `jwt` handler configuration for Ory `jwt` handler
```
Comment (barchw) [inline docs/04-operation-guides/troubleshooting/api-exposure/apix-10-api-rule-troubleshooting.md:131]: I assume that we shouldn't mention Istio JWT (yet). Maybe we should update this troubleshooting guide with configuration errors for handlers when we will announce the feature for customers?

Comment (videlov) [reply docs/04-operation-guides/troubleshooting/api-exposure/apix-10-api-rule-troubleshooting.md:131]: This is a good point. @strekm  what do you think? Should we wait for Istio `jwt` feature or first make this troubleshooting guide for non-Istio `jwt` cases and extend afterwards?

**Review by barchw — CHANGES_REQUESTED** [review]

File: docs/04-operation-guides/troubleshooting/api-exposure/apix-10-api-rule-troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+      accessStrategies:
+        - handler: jwt
+          config:
+            authentications:
+            - issuer: "https://example.com"
+              jwksUri: "https://example.com/.well-known/jwks.json"
+```
+
+You should always refer to the technical reference documentation for the [APIRule CR](https://kyma-project.io/docs/kyma/latest/05-technical-reference/00-custom-resources/apix-01-apirule/)
+
```
[the remark this answers is not in the document]
Comment (videlov) [reply docs/04-operation-guides/troubleshooting/api-exposure/apix-10-api-rule-troubleshooting.md:245]: yes this makes sense. perhaps is better to reference to the value.yaml and not copy its values here.

File: docs/04-operation-guides/troubleshooting/api-exposure/apix-10-api-rule-troubleshooting.md (hunk 1 above)
Comment (strekm) [reply docs/04-operation-guides/troubleshooting/api-exposure/apix-10-api-rule-troubleshooting.md:131]: @videlov yes, please remove this point for now

File: docs/04-operation-guides/troubleshooting/api-exposure/apix-10-api-rule-troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+        - handler: jwt
+          config:
+            trusted_issuers: ["https://example.com"]
+            jwks_urls: ["https://example.com/.well-known/jwks.json"]
+```
+
+You should always refer to the technical reference documentation for the [APIRule CR](https://kyma-project.io/docs/kyma/latest/05-technical-reference/00-custom-resources/apix-01-apirule/)
+
+---
+## Issue - Using Ory `jwt` handler configuration for Istio `jwt` handler
```
Comment (strekm) [inline docs/04-operation-guides/troubleshooting/api-exposure/apix-10-api-rule-troubleshooting.md:189]: this should be removed as well for now.

File: docs/04-operation-guides/troubleshooting/api-exposure/apix-10-api-rule-troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+---
+## Issue - Service in APIRule is blocklisted
+
+If your APIRule specifies a service that is blocklisted, e.g.:
+
+```yaml
+spec:
+  ...
+  service:
+    name: apiserver-proxy
```
Comment (strekm) [inline docs/04-operation-guides/troubleshooting/api-exposure/apix-10-api-rule-troubleshooting.md:255]: apiserver-proxy is component that was removed with kyma 2. we should update our configuration (follow up) and not use it as an example of blocklisted service.  one of istio services might be good example

File: docs/04-operation-guides/troubleshooting/api-exposure/apix-10-api-rule-troubleshooting.md (hunk 3 above)
Comment (videlov) [reply docs/04-operation-guides/troubleshooting/api-exposure/apix-10-api-rule-troubleshooting.md]: removed

**Review by barchw — APPROVED** [review]

**Review by nataliasitko — APPROVED** [review]

**Pull request merged**